tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/core/waypoint.h

```diff
@@ -62,7 +68,12 @@ struct WaypointInnerBase
 template <typename T>
 struct WaypointInner final : WaypointInnerBase
 {
-  WaypointInner() = default;
+  WaypointInner()
+  {
+    static_assert(has_member_Waypoint_getType<T>::value, "Class does not have member function 'getType'");
+    static_assert(has_member_Waypoint_print<T>::value, "Class does not have member function 'print'");
+    static_assert(has_member_Waypoint_toXML<T>::value, "Class does not have member function 'has_member_toXML'");
```

Comment:
typo: `Class does not have member function 'has_member_toXML `should be `Class does not have member function 'toXML'`

Reply:
Yea, not sure how to create a test for something like this where it causes compiler failures; any thoughts? I did tested it out in godbolt to verify that it works.

Reply:
I think I have a solution for a unit test. Testing now.

Reply:
Oh I just meant make 2 test classes and see if checking for that method works

Reply:
Yea, that is what I did. In the process I did some testing with Instruction and Waypoint and realized I forgot to add the static asserts to the move and copy constructor. In addition I think there is also a way to leverage enable_if_t in the template definition, but not sure how to handle multiple conditions. Though I like the static assert because you get a meaningful message but you have to make sure to add them to all available constructors where the other automatically works for all.

``` c++
template <typename T, std::enable_if_t<has_member_getType<T>::value, bool> = true>
```